src/hooks/usePoolTickData.ts

```diff
@@ -0,0 +1,109 @@
+import { Currency } from '@uniswap/sdk-core'
+import { FeeAmount, nearestUsableTick, TickMath, tickToPrice, TICK_SPACINGS } from '@uniswap/v3-sdk'
+import keyBy from 'lodash.keyby'
+import JSBI from 'jsbi'
+import { PoolState, usePool } from './usePools'
+import { useAllV3Ticks } from './useAllV3Ticks'
+import { useMemo } from 'react'
+import { TickProcessed } from 'constants/ticks'
+import computeSurroundingTicks from 'utils/computeSurroundingTicks'
+
+const PRICE_FIXED_DIGITS = 8
+
+export function usePoolTickData(
+  currencyA: Currency | undefined,
+  currencyB: Currency | undefined,
+  feeAmount: FeeAmount | undefined
+): {
+  loading: boolean
+  error: boolean
+  valid: boolean
+  activeTick: number | undefined
+  tickData: TickProcessed[]
+} {
+  const pool = usePool(currencyA, currencyB, feeAmount)
+
+  const tickSpacing = feeAmount && TICK_SPACINGS[feeAmount]
+
+  // Find nearest valid tick for pool in case tick is not initialized.
+  const activeTick =
```

Comment:
hmm, should this be rounded down? if `tickCurrent` is 9 (with a tick spacing of 10), i'd define 0 as the nearest "active" tick, because liquidity is active at ticks >= lower but < upper

Reply:
is there a util to round down based on tick spacing?

Reply:
not as far as i know, but subtracting tick mod tickSpacing should probably do the trick